nmigen_boards/dev/spi.py

```diff
@@ -4,16 +4,25 @@
 __all__ = ["SPIResource"]
 
 
-def SPIResource(number, *, cs, clk, mosi, miso, int=None, reset=None, attrs=None):
+def SPIResource(number, *, cs, clk, mosi, miso, int=None, reset=None, attrs=None, name="spi", role="host"):
+    def get_dir(host_out=True):
+        if host_out:
+            return "o" if role == "host" else "i"
+        else:
+            return "i" if role == "host" else "o"
+        
+    if role not in ["host", "device"]:
+        raise ValueError("Argument \"role\" must be \"host\" or \"device\", not {}.".format(role))
+    
     io = []
-    io.append(Subsignal("cs", PinsN(cs, dir="o")))
-    io.append(Subsignal("clk", Pins(clk, dir="o")))
-    io.append(Subsignal("mosi", Pins(mosi, dir="o")))
-    io.append(Subsignal("miso", Pins(miso, dir="i")))
-    if int is not None:
+    io.append(Subsignal("cs", PinsN(cs, dir=get_dir())))
+    io.append(Subsignal("clk", Pins(clk, dir=get_dir())))
+    io.append(Subsignal("mosi", Pins(mosi, dir=get_dir())))
+    io.append(Subsignal("miso", Pins(miso, dir=get_dir(False))))
```

Comment:
Never, ever, use non-keyword boolean arguments unless the name of the function serves as the name of the argument as well. Don't do it in languages without keyword arguments either... in C++ something like `/*host_out=*/false` works.

Anyway, it's also pointlessly complex to make this a function. Just inline the logic.